src/monitors/WhammMonitor.v3

```diff
@@ -83,8 +83,83 @@ class WhammMonitor(whamm: Module) extends Monitor {
 		for (l = bp; l != null; l = l.tail) { // for every probe
 			if (Trace.whamm) Trace.OUT.put3("  match func %q+%d: %s", func.render(module.names, _), opcode_pos, opcode.mnemonic).ln();
 			def wf   = wi.functions[bp.head.2.func_index];
-			def args = bindParams(cv, wf, wi, bp.head.1, opcode, false);
-			if (args != null) Instrumentation.insertLocalProbe(module, func.func_index, opcode_pos, WhammProbe.new(wf, args));
+			// check static predicate
+			if (evalPredicate(cv, wi, opcode, bp.head.3)) {
+				def args = bindParams(cv, wf, wi, bp.head.1, opcode, false);
+				if (args != null) Instrumentation.insertLocalProbe(module, func.func_index, opcode_pos, WhammProbe.new(wf, args));
+			}
+		}
+	}
+
+	def evalPredicate(cv: CodeValidator, wi: Instance, op: Opcode, pred: WhammPredicate) -> bool {
+		match (pred) {
+			True => return true;
+			WhammParam(param) => match (param) {
+				Call(target, params) => {
+					def exports = wi.module.exports;
+					def lookupR = FuncDecl.!(lookup(exports, target.image, Strings.equal, null));
+					if (lookupR == null) {
+						err.at(cv.parser.decoder).WhammProbeError("whamm probe", "function in the probe argument not found");
+						return false;
+					}
+
+					def index = lookupR.func_index;
+					def func = wi.functions[index];
+
+					def whammArgs = bindParams(cv, WasmFunction.!(func), wi, params, op, true);
+					if (whammArgs == null) return false;	// propagating error, not a good idea
+					def args = if(whammArgs.length == 0, Values.NONE, Array<Value>.new(whammArgs.length));
+
+					for (i < whammArgs.length) {
+						var v: Value;
+						match (whammArgs[i]) {
```

Comment:
Seems like a good candidate to factor out the commonalities with a `WhammProbe`'s invocation.